sig-node/openharmony/implementation/KubeEdge Runs On OpenHarmony.md

```diff
@@ -0,0 +1,1277 @@
+mount --bind /data/opt /opt
+mount --bind /data/usr /usr
+mount --bind /data/bin /bin
+mount --bind /data/root /root
+#mount cgroup
+mount -t cgroup -o none,name=systemd cgroup /sys/fs/cgroup/systemd
+mount -t cgroup -o blkio,nodev,noexec,nosuid cgroup /sys/fs/cgroup/blkio
+mount -t cgroup -o cpu,nodev,noexec,nosuid cgroup /sys/fs/cgroup/cpu
+mount -t cgroup -o cpuacct,nodev,noexec,nosuid cgroup /sys/fs/cgroup/cpuacct
+mount -t cgroup -o cpuset,nodev,noexec,nosuid cgroup /sys/fs/cgroup/cpuset
+mount -t cgroup -o devices,nodev,noexec,nosuid cgroup /sys/fs/cgroup/devices
+mount -t cgroup -o freezer,nodev,noexec,nosuid cgroup /sys/fs/cgroup/freezer
+mount -t cgroup -o hugetlb,nodev,noexec,nosuid cgroup /sys/fs/cgroup/hugetlb
+mount -t cgroup -o memory,nodev,noexec,nosuid cgroup /sys/fs/cgroup/memory
+mount -t cgroup -o net_cls,nodev,noexec,nosuid cgroup /sys/fs/cgroup/net_cls
+mount -t cgroup -o net_prio,nodev,noexec,nosuid cgroup /sys/fs/cgroup/net_prio
+mount -t cgroup -o perf_event,nodev,noexec,nosuid cgroup /sys/fs/cgroup/perf_event
+mount -t cgroup -o pids,nodev,noexec,nosuid cgroup /sys/fs/cgroup/pids
+```
+
+- Add some route rules
+
+```
+ip rule add pref 1 from all lookup main
+ip rule add pref 2 from all lookup default
+```
+
+- Close the selinux
+
+```
+setenforce 0
+```
+
+- Create file /etc/dock/daemon. json and write
+
+```
+{"registry-mirrors":["https://docker.mirrors.ustc.edu.cn"],"experimental":true}
+```
+
+- Run Docker
+
+```
+dockerd -H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock & 
+```
+
+- Verify docker working status
+
+```
+# Ensure the OpenHarmony is networked then execute hello-world
+docker run hello-world
+
+# You can see
+Unable to find image 'hello-world:latest' locally
+latest: Pulling from library/hello-world
+93288797bd35: Pull complete 
+Digest: sha256:2498fce14358aa50ead0cc6c19990fc6ff866ce72aeb5546e1d59caac3d0d60f
+Status: Downloaded newer image for hello-world:latest
+
+Hello from Docker!
+This message shows that your installation appears to be working correctly.
+
+To generate this message, Docker took the following steps:
+ 1. The Docker client contacted the Docker daemon.
+ 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
+    (arm64v8)
+ 3. The Docker daemon created a new container from that image which runs the
+    executable that produces the output you are currently reading.
+ 4. The Docker daemon streamed that output to the Docker client, which sent it
+    to your terminal.
+
+To try something more ambitious, you can run an Ubuntu container with:
+ $ docker run -it ubuntu bash
+
+Share images, automate workflows, and more with a free Docker ID:
+ https://hub.docker.com/
+
+For more examples and ideas, visit:
+ https://docs.docker.com/get-started/
+```
+
+
+
+## Key steps for OpenHarmony to run KubeEdge
+
+
+
+Use KubeEdge to bridge the cloud and the OpenHarmony edge to achieve cloud-side collaboration.
+
+
+
+##### 1. Install KubeEdge cloudcore on the cloud and get token
```

Comment:
Suggest： 
 #### 1. Install K8s 
 #### 2. Deploy KubeEdge's cloudcore with keadm
keadm init xxx (wirte key step here)
#### 3. KubeEdge edgecore source static compilation
#### 4. Install edgecore on the OpenHarmony edge